packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -84,144 +108,124 @@ public void onCancel(Object o) {
             }
           });
 
-      mediaPlayer.setSurface(new Surface(textureEntry.surfaceTexture()));
-      setAudioAttributes(mediaPlayer);
-      mediaPlayer.setOnPreparedListener(
-          new MediaPlayer.OnPreparedListener() {
+      surface = new Surface(textureEntry.surfaceTexture());
+      exoPlayer.setVideoSurface(surface);
+      exoPlayer.setPlayWhenReady(true);
+
+      exoPlayer.addListener(
+          new EventListener() {
```

Comment:
In case you missed my earlier comment here (we had a mid-air collision with your white space changes): please consider subclassing `DefaultEventListener` instead to avoid the empty methods.